### ISSUE TITLE
sbt 1.3.2 and above fail to run with `java.lang.BootstrapMethodError: java.lang.NoClassDefFoundError: scala/runtime/java8`

### ISSUE THREAD

**Issue by kiara-riley** [sbt/sbt#5231]
## steps
sbt version: 1.3.3 and 1.3.2

1. Install `sbt` and run in empty directory

## problem
```
$ sbt
java.lang.BootstrapMethodError: java.lang.NoClassDefFoundError: scala/runtime/java8/JFunction0$mcZ$sp
        at sbt.internal.SysProp$.getOrFalse(SysProp.scala:30)
        at sbt.internal.SysProp$.client(SysProp.scala:70)
        at sbt.xMain$.run(Main.scala:51)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sbt.internal.XMainConfiguration.run(XMainConfiguration.scala:45)
        at sbt.xMain.run(Main.scala:39)
        at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:111)
        at xsbt.boot.Launch$.withContextLoader(Launch.scala:130)
        at xsbt.boot.Launch$.run(Launch.scala:111)
        at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:37)
        at xsbt.boot.Launch$.launch(Launch.scala:119)
        at xsbt.boot.Launch$.apply(Launch.scala:20)
        at xsbt.boot.Boot$.runImpl(Boot.scala:56)
        at xsbt.boot.Boot$.main(Boot.scala:18)
        at xsbt.boot.Boot.main(Boot.scala)
Caused by: java.lang.NoClassDefFoundError: scala/runtime/java8/JFunction0$mcZ$sp
        ... 18 more
Caused by: java.lang.ClassNotFoundException: scala.runtime.java8.JFunction0$mcZ$sp
        at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
        ... 18 more
[error] [launcher] error during sbt launcher: java.lang.BootstrapMethodError: java.lang.NoClassDefFoundError: scala/runtime/java8/JFunction0$mcZ$sp
```

## expectation

sbt loads

## notes
Installed using bintray yum repo: https://www.scala-sbt.org/1.x/docs/Installing-sbt-on-Linux.html

OS:
```
$ cat /etc/fedora-release 
Fedora release 30 (Thirty)
```

Install:
```
$ sudo dnf info sbt
Last metadata expiration check: 0:03:44 ago on Tue 12 Nov 2019 01:07:45 PM MST.
Installed Packages
Name         : sbt
Version      : 1.3.3
Release      : 0
Architecture : noarch
Size         : 1.4 M
Source       : sbt-1.3.3-0.src.rpm
Repository   : @System
From repo    : bintray--sbt-rpm
Summary      : sbt, the interactive build tool
URL          : http://github.com/sbt/sbt-launcher-package
License      : BSD
Description  : This script provides a native way to run sbt,
             :   a build tool for Scala and more.
```

Current java version(newest in repos):
```
$ java -version
openjdk version "1.8.0_232"
OpenJDK Runtime Environment (build 1.8.0_232-b09)
OpenJDK 64-Bit Server VM (build 25.232-b09, mixed mode)
```

Current Dir:
```
ls -al
total 16
drwxrwxr-x.  3 kmckernan kmckernan 4096 Nov 12 13:04 .
drwxrwxr-x. 26 kmckernan kmckernan 4096 Nov 12 13:04 ..
drwxrwxr-x.  8 kmckernan kmckernan 4096 Nov 12 13:05 .git
-rw-rw-r--.  1 kmckernan kmckernan   19 Nov 12 13:04 README.md
```

----
I am assuming this is a problem in 1.3.2 and above since I can install 1.3.1 and it runs:
```
$ sbt
[residual] arg = '-sbt-create'
[residual] arg = '-sbt-create'
[process_args] java_version = '8'
[sbt_options] declare -a sbt_options=()
[addMemory] arg = '1024'
[addJava] arg = '-Xms1024m'
[addJava] arg = '-Xmx1024m'
[addJava] arg = '-Xss4M'
[addJava] arg = '-XX:ReservedCodeCacheSize=128m'
# Executing command line:
java
-Dfile.encoding=UTF-8
-Xms1024m
-Xmx1024m
-Xss4M
-XX:ReservedCodeCacheSize=128m
-jar
/usr/share/sbt/bin/sbt-launch.jar

[warn] No sbt.version set in project/build.properties, base directory: /home/kmckernan/git/sbt-nexus-example
[info] Loading settings for project global-plugins from build.sbt ...
[info] Loading global plugins from /home/kmckernan/.sbt/1.0/plugins
[info] Set current project to sbt-nexus-example (in build file:/home/kmckernan/git/sbt-nexus-example/)
[info] sbt server started at local:///home/kmckernan/.sbt/1.0/server/0e82540cb21007fc065d/sock
sbt:sbt-nexus-example>
```

1.3.2 is slightly different but has the same exception as 1.3.3:
```
$ sbt
Exception in thread "sbt-parser-init-thread" java.lang.NoSuchMethodError: scala.Some.value()Ljava/lang/Object;
        at sbt.io.IO$.uriToFile(IO.scala:257)
        at sbt.io.IO$.toFile(IO.scala:227)
        at sbt.io.IO$.classLocationPath(IO.scala:67)
        at sbt.io.IO$.classLocationPath(IO.scala:81)
        at sbt.internal.parser.SbtParser$.<init>(SbtParser.scala:45)
        at sbt.internal.parser.SbtParser$.<clinit>(SbtParser.scala)
        at sbt.internal.parser.SbtParserInit$$anon$2.run(SbtParser.scala:171)
java.lang.BootstrapMethodError: java.lang.NoClassDefFoundError: scala/runtime/java8/JFunction0$mcZ$sp
        at sbt.internal.SysProp$.getOrFalse(SysProp.scala:30)
        at sbt.internal.SysProp$.client(SysProp.scala:70)
        at sbt.xMain$.run(Main.scala:51)
        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.lang.reflect.Method.invoke(Method.java:498)
        at sbt.internal.XMainConfiguration.run(XMainConfiguration.scala:46)
        at sbt.xMain.run(Main.scala:39)
        at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:111)
        at xsbt.boot.Launch$.withContextLoader(Launch.scala:130)
        at xsbt.boot.Launch$.run(Launch.scala:111)
        at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:37)
        at xsbt.boot.Launch$.launch(Launch.scala:119)
        at xsbt.boot.Launch$.apply(Launch.scala:20)
        at xsbt.boot.Boot$.runImpl(Boot.scala:56)
        at xsbt.boot.Boot$.main(Boot.scala:18)
        at xsbt.boot.Boot.main(Boot.scala)
Caused by: java.lang.NoClassDefFoundError: scala/runtime/java8/JFunction0$mcZ$sp
        ... 18 more
Caused by: java.lang.ClassNotFoundException: scala.runtime.java8.JFunction0$mcZ$sp
        at java.net.URLClassLoader.findClass(URLClassLoader.java:382)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:418)
        at java.lang.ClassLoader.loadClass(ClassLoader.java:351)
        ... 18 more
[error] [launcher] error during sbt launcher: java.lang.BootstrapMethodError: java.lang.NoClassDefFoundError: scala/runtime/java8/JFunction0$mcZ$sp

```

I don't see any notes in scala https://github.com/scala/scala/releases/tag/v2.12.10 that seem to cause this or in sbt https://github.com/sbt/sbt/releases/tag/v1.3.2

**Comment by eed3si9n** [comment]
The steps works for me on macOS:

```
$ sbt
[warn] No sbt.version set in project/build.properties, base directory: /private/tmp/test
[info] Loading settings for project global-plugins from idea.sbt,vimquit.sbt,plugins.sbt,metals.sbt,pgp.sbt ...
[info] Loading global plugins from /Users/eed3si9n/.sbt/1.0/plugins
[info] Set current project to test (in build file:/private/tmp/test/)
[info] sbt server started at local:///Users/eed3si9n/.sbt/1.0/server/e64cee8b21620f4bb923/sock
sbt:test> exit
```

What version of Java are you using?

**Comment by kiara-riley** [comment]
`1.8.0_232` latest fedora 30 repos had at time of posting.

**Comment by quizzicus** [comment]
Problem still exists in sbt 1.3.6.

**Comment by seandockery** [comment]
Is it possible that this is distro-specific?  Does it affect distros other than Fedora?  Does anyone know?

**Comment by seandockery** [comment]
We are seeing the same behaviour with sbt 1.3.8, Fedora 31, and Java 1.8.0_232.

**Comment by seandockery** [comment]
Error while importing sbt project:

```
List(java.lang.BootstrapMethodError: java.lang.NoClassDefFoundError: scala/runtime/java8/JFunction0$mcZ$sp
at sbt.internal.SysProp$.getOrFalse(SysProp.scala:30)
at sbt.internal.SysProp$.client(SysProp.scala:70)
at sbt.xMain$.run(Main.scala:51)
at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
at java.lang.reflect.Method.invoke(Method.java:498)
at sbt.internal.XMainConfiguration.run(XMainConfiguration.scala:45)
at sbt.xMain.run(Main.scala:39)
at xsbt.boot.Launch$$anonfun$run$1.apply(Launch.scala:109)
at xsbt.boot.Launch$.withContextLoader(Launch.scala:128)
at xsbt.boot.Launch$.run(Launch.scala:109)
at xsbt.boot.Launch$$anonfun$apply$1.apply(Launch.scala:35)
at xsbt.boot.Launch$.launch(Launch.scala:117)
at xsbt.boot.Launch$.apply(Launch.scala:18)
at xsbt.boot.Boot$.runImpl(Boot.scala:56)
at xsbt.boot.Boot$.main(Boot.scala:18)
at xsbt.boot.Boot.main(Boot.scala)
```

**Comment by eed3si9n** [comment]
I tried using Docker and Fedora 31 and it works for me:

```
# java -version
openjdk version "1.8.0_242"
OpenJDK Runtime Environment (build 1.8.0_242-b08)
OpenJDK 64-Bit Server VM (build 25.242-b08, mixed mode)

# sbt
[info] [launcher] getting org.scala-sbt sbt 1.3.8  (this may take some time)...
downloading https://repo1.maven.org/maven2/org/scala-sbt/sbt/1.3.8/sbt-1.3.8.jar ...
:: loading settings :: url = jar:file:/usr/share/sbt/bin/sbt-launch.jar!/org/apache/ivy/core/settings/ivysettings.xml
downloading https://repo1.maven.org/maven2/org/scala-lang/scala-library/2.12.10/scala-library-2.12.10.jar ...
downloading https://repo1.maven.org/maven2/org/scala-sbt/io_2.12/1.3.3/io_2.12-1.3.3.jar ...
:: loading settings :: url = jar:file:/usr/share/sbt/bin/sbt-launch.jar!/org/apache/ivy/core/settings/ivysettings.xml
:: loading settings :: url = jar:file:/usr/share/sbt/bin/sbt-launch.jar!/org/apache/ivy/core/settings/ivysettings.xml
downloading https://repo1.maven.org/maven2/org/scala-sbt/main_2.12/1.3.8/main_2.12-1.3.8.jar ...
:: loading settings :: url = jar:file:/usr/share/sbt/bin/sbt-launch.jar!/org/apache/ivy/core/settings/ivysettings.xml
	[SUCCESSFUL ] org.scala-sbt#sbt;1.3.8!sbt.jar (246ms)
downloading https://repo1.maven.org/maven2/org/scala-sbt/logic_2.12/1.3.8/logic_2.12-1.3.8.jar ...
....
:: retrieving :: org.scala-sbt#boot-scala
	confs: [default]
	6 artifacts copied, 0 already retrieved
[warn] No sbt.version set in project/build.properties, base directory: /
[info] Set current project to root (in build file:/)
[info] sbt server started at local:///root/.sbt/1.0/server/e656f0eb572233bacf5f/sock
sbt:root> exit
[info] shutting down sbt server
```

Here are the commands in the history:

```
# history
yum install java-1.8.0-openjdk-devel
curl https://bintray.com/sbt/rpm/rpm | tee /etc/yum.repos.d/bintray-sbt-rpm.repo
yum install sbt
sbt
java -version
```

**Comment by vlmiroshnikov** [comment]
I  fix the same problem afted deleting `.sbt`.

```
cat /etc/redhat-release 
Fedora release 30 (Thirty)
```

```
java -version 
openjdk version "1.8.0_242"
OpenJDK Runtime Environment (build 1.8.0_242-b08)
OpenJDK 64-Bit Server VM (build 25.242-b08, mixed mode)
```

`sbt script version: 1.3.8`


This problem was appear after upgrade fedora from 28 to 30

**Comment by hochgi** [comment]
also happened to me on fedora.

But in my case it was probably intelliJ corrupting the `~/.sbt` cache.
Everything worked fine for quite some time.
I had multiple scala projects opened.
In one of them, IntelliJ insisted that scala version was 2.12 wrongly (it was 2.13).
I tried everything.
Eventually, from within IntelliJ I removed "External Libraries" wrong scala sources,
invalidated caches and restart, and then the problem manifested.

After using @vlmiroshnikov's solution, it all worked again.
